src/solidity-frontend/solidity_grammar.h

```diff
@@ -67,6 +101,24 @@ enum ElementaryTypeNameT
 };
 ElementaryTypeNameT get_elementary_type_name_t(const nlohmann::json &type_name);
 const char *elementary_type_name_to_str(ElementaryTypeNameT type);
+static std::map<ElementaryTypeNameT, unsigned int> uintSizeMap = {
```

Comment:
While I understand exporting a way to map from the enum to the type might be necessary, making this static declaration in a header file is not a good approach. I have two issues with this:
1. not every mapping needs to be a `std::map`; as the enum is layed out, this could just as well be an array. Or a switch.
2. Having this `static` in the header means that every translation unit that #include-s it also gets a local `uintSizeMap` symbol. That is wasteful.

So, the suggestion is to move this symbol declaration to the place where it is needed and if exporting a way to obtain the size for one of these enum values is required, to add a function for that.

Also: if you keep the std::map, can it be `const`?